test/utils/createRenderer.tsx

```diff
@@ -513,6 +513,9 @@ export function createRenderer(globalOptions: CreateRendererOptions = {}): Rende
     profiler = new UsedProfiler(test);
 
     emotionCache = createEmotionCache({ key: 'emotion-client-render' });
+    // To disable SSR warnings when using CSS pseudo selectors, e.g. :first-child
+    // https://github.com/emotion-js/emotion/issues/1105#issuecomment-1058225197
+    emotionCache.compat = true;
```

Comment:
I would expect that we can't have this logic. Many developers don't inline their CSS in the `<head>` (like we recommend) but let emotion render them in the `<body>`. If we need this rule here, it means that we broke support for the setup of SSR without configuration.

```suggestion
```

Reply:
I thought that we don't support zero-config SSR (https://mui.com/material-ui/guides/server-rendering/).

In my opinion, we must not support zero-config SSR. It's a very bad experience to not be able to write plain CSS like `:first-child` 🥲 (note that `:first-of-type` is not a replacement, it is for different use cases). cc @mui/core

Reply:
I agree with the theory, but in practice, it's hard to get this zero-config not support fly.

Reply:
This is just about tests. The table still supports zero-config SSR since the children of the table are plain HTML elements.

I will update the tests to add compat as an option just to make the tests pass.